src/test/run-make-fulldeps/coverage-reports-deadcode/expected_show_coverage.assert.txt

```diff
@@ -0,0 +1,34 @@
+    1|       |#![allow(unused_assignments)]
```

Comment:
I'm not seeing how these -deadcode test files are any different, or add value. I guess that's one piece of the cleanup to be done.

Reply:
The `-Clink-dead-code` defaults to `on` on Linux and Mac, and `off` on Windows. If the results are different, the tests would fail on Windows if I only had one variation.

I wasn't sure if I'd get different coverage results depending on the flag. Now that I have several examples, I'll `diff` them to see. It's a little risky to assume the flag makes no difference, even if true for the current tests.

Reply:
@tmandry - FYI, currently, two tests have slightly different coverage results.

```shell
$ diff -r coverage-reports-base coverage-reports-deadcode -x '*counters*' -x '*.json' -x 'prett*' -x 'Make*'
```
```diff
15c15
<    15|       |async fn e() -> u8 { 1 }
---
>    15|      0|async fn e() -> u8 { 1 }
19c19
<    19|       |async fn foo() -> [bool; 10] { [false; 10] }
---
>    19|      0|async fn foo() -> [bool; 10] { [false; 10] }
diff --color -r -x '*counters*' -x '*.json' -x 'prett*' -x 'Make*' '--color=never' coverage-reports-base/expected_show_coverage.partial_eq.txt coverage-reports-deadcode/expected_show_coverage.partial_eq.txt
5a6,15
>   ------------------
>   | Unexecuted instantiation: <partial_eq::Version as core::cmp::PartialOrd>::gt
>   ------------------
>   | Unexecuted instantiation: <partial_eq::Version as core::cmp::PartialOrd>::le
>   ------------------
>   | Unexecuted instantiation: <partial_eq::Version as core::cmp::PartialOrd>::ge
>   ------------------
>   | <partial_eq::Version as core::cmp::PartialOrd>::lt:
>   |    4|      1|#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
>   ------------------
8a19,28
>   ------------------
>   | Unexecuted instantiation: <partial_eq::Version as core::cmp::PartialOrd>::gt::{closure#0}
>   ------------------
>   | Unexecuted instantiation: <partial_eq::Version as core::cmp::PartialOrd>::ge::{closure#0}
>   ------------------
>   | <partial_eq::Version as core::cmp::PartialOrd>::lt::{closure#0}:
>   |    7|      1|    minor: usize,
>   ------------------
>   | Unexecuted instantiation: <partial_eq::Version as core::cmp::PartialOrd>::le::{closure#0}
>   ------------------
```

Reply:
deadcode is no longer applicable to coverage! I'm able to instrument the deadcode functions without using the `-Clink-dead-code` option. Special tests are now gone.